### DISCUSSION TITLE
Please migrate your code to use AWS SDK for JavaScript (v3).

### DISCUSSION THREAD

**Discussion by lunarballoon** [serverless/serverless#11750] [General]
everytime i run a serverless command i keep getting this note

(node:55677) NOTE: The AWS SDK for JavaScript (v2) will be put into maintenance mode in 2023.

Please migrate your code to use AWS SDK for JavaScript (v3).
For more information, check the migration guide at https://a.co/7PzMCcy
(Use `node --trace-warnings ...` to show where the warning was created)

is there anyway to stop it from outputing this?

**Comment by alexeyshockov** [comment]
Can confirm, it is very annoying

**Comment by Zhuravlev1** [comment]
I have the same issue. Moreover, it's very strange because my project uses AWS SDK V3.

**Reply by vicary** [reply]
It comes from Serverless itself, they need to upgrade it.

**Reply by StevenWang20230313** [reply]
Hi Vicary, do you mean upgrading serverless framework to V3. The link https://www.serverless.com/framework/docs/guides/upgrading-v3

**Reply by vicary** [reply]
@StevenWang20230313 As mentioned in this very same discussion https://github.com/serverless/serverless/discussions/11750#discussioncomment-5583571.

It's indirectly introduced in Serverless v3 because `aws-sdk` added that in a minor, but it is then suppressed in v3.28.1.

**Reply by StevenWang20230313** [reply]
Confused. For my repo, I used AWS official package to update the code automatically and manually searched the entire repo to make some changes on v2 code. I think I made all the changes for v3 migration. But, I still have that message to ask me to migrate my code. Does that because of serverless framework? Should I upgrade serverless framework to V3?

**Reply by vicary** [reply]
Upgrading your direct dependency of aws-sdk to v3 does nothing, because it's serverless which depends on the noisy aws-sdk v2.

Try upgrading to the latest version of serverless v3.

**Reply by StevenWang20230313** [reply]
Thank you very much.

**Comment by Bandes** [comment]
Thank you for posting this, I'm having the same issue and I thought it was just me!

**Comment by asv0018** [comment]
Oh my god!!!!!!, even am facing this issue...

**Comment by kumarasinghe** [comment]
Seeing this in `aws-nodejs-typescript` template as well

**Comment by MohamedDhiaZoghlami** [comment]
Same goes here its been 2 days im having this issue and i even started a new project to test and i used v3 yet same thing

**Comment by jrewcastle** [comment]
What PATH are they referencing in **npx aws-sdk-js-codemod -t v2-to-v3 PATH**

**Reply by LALIT-SEKHAR** [reply]
i thing the file you want to migrate the  code

**Comment by DavidWells** [comment]
"Fixed this" by pinning aws-sdk from a week ago before deprecation notice. 

https://github.com/serverless/serverless/issues/11753#issuecomment-1433733188

Looking forward to probably using this https://github.com/sam-goodwin/itty-aws in the future

**Comment by cgonc** [comment]
It is really annoying and triggers my OCD. Please help me I am having a panic attack.

**Reply by vamsikrishnamathala** [reply]
It is just a warning migrate to V3 and use code below in the file where aws sdk is being initialized.
`require('aws-sdk/lib/maintenance_mode_message').suppress = true;`
[https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/migrating-to-v3.html](url)

**Reply by khue20** [reply]
work for me thank man

**Reply by kiran-cj** [reply]
worked for me too
Thanks buddy!

**Comment by eapariciodev** [comment]
You can suppress the message by adding this code:

```
require('aws-sdk/lib/maintenance_mode_message').suppress = true;
```

It is noted in the code that displays the warning, https://github.com/aws/aws-sdk-js/blob/master/lib/maintenance_mode_message.js

**Reply by vicary** [reply]
Where do I put this line to make sure serverless CLI picks it up when I run any commands?

**Reply by arshianas** [reply]
can you please tell the LOCATION where to put this line?

**Reply by eapariciodev** [reply]
Apologies, I stumbled on this thread while searching for a solution and didn't catch that this was for CLI commands. I was looking to remove the warnings when using `serverless-offline` while developing/testing, so I was able to add the suppress line to my own code.

I'm not sure that this is really a `serverless` issue. The aws-sdk dependency they use is an older version that doesn't have the warning yet. For me it was the `serverless-domain-manager` plugin that was displaying the warning. I was able to suppress it by adding the line into that package's index.js file. I guess you'd have to figure out which plugin(s) are displaying the warning for you.

**Reply by vicary** [reply]
The warning was introduced in aws-sdk 2.1313.0, and Serverless depends on `^2.1303.0` which means a fresh installation or lock file update should already gives you this warning.

**Comment by et304383** [comment]
Since no one seems to have complained to the AWS SDK team at AWS, I created an issue:

https://github.com/aws/aws-sdk-js/issues/4354

**Comment by ibiren** [comment]
Migrating to **NodeJS18.x** and **AWS-SDK-v3** will reduce package size by 90% and improve performance by 75%. This should be done on priority. 
https://aws.amazon.com/blogs/developer/modular-packages-in-aws-sdk-for-javascript/

**Reply by vicary** [reply]
If users are using the bundled version of `aws-sdk`, 90% package size reduction means little since the multiplier is 0.

**Reply by Zhuravlev1** [reply]
Be careful to migrate to node v18 such libraries like sharp, puppeter will break(at least I can not use them in lambdas). Also, you don't need to pack Aws SDK because it's already in AWS.

**Reply by vicary** [reply]
@Zhuravlev1 Try `@sparticuz/chromium`, it works in 18.x.

**Comment by zakdaff** [comment]
Hello everyone, I have encountered the same error. Is it necessary to migrate to v3??  Thanks for your help

**Comment by daniel-pledge-io** [comment]
Serverless v3.28.1 suppresses the warning, see https://github.com/serverless/serverless/releases/tag/v3.28.1

**Reply by zakdaff** [reply]
Thanks a lot @daniel-pledge-io

**Comment by alextello** [comment]
in my case, i'm using s3, so i move from aws-sdk to @aws-sdk/client-s3 and that solve my problem.

**Comment by DarlonHenrique** [comment]
same thing here, im trying to use turbo run dev and turbo run build but i keep receiving this

**Comment by h-evers** [comment]
There are options mentioned in the docs to suppress the output:
https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/

`process.env.AWS_SDK_JS_SUPPRESS_MAINTENANCE_MODE_MESSAGE = '1'`

respective

`AWS_SDK_JS_SUPPRESS_MAINTENANCE_MODE_MESSAGE=1 node my_program.js`

**Comment by 10Bude10** [comment]


**Reply by vicary** [reply]
It almost makes sense if aws-sdk means V2. If you mean they should upgrade to V3 then yes please!

**Comment by SunitaMArshall** [comment]
Thank you for posting this, I'm having the same issue and I thought it was just me. It is also very [annoying](https://statuskisan.com/pm-kisan-ekyc-status/)!

**Comment by czubocha** [comment]
Support for AWS SDK for JavaScript v3 is available starting in [v4.19.0](https://github.com/serverless/serverless/releases/tag/v4.19.0). You can enable it by setting the environment variable `SLS_AWS_SDK=3`. See the [docs](https://www.serverless.com/framework/docs/providers/aws/guide/deploying#aws-sdk-for-javascript-v3) for more details.
